### PR TITLE
Implement new ckb-combine-lock

### PR DESCRIPTION
**Pull request by mohanson**
Type the following command to test the new combine-lock

```
$ make -C ckb-debugger-tests cl-always-success-v2
```

### CODE REVIEW

**Comment by XuJiandong** [comment]
We don't need "v2": just replace origina one.

**Comment by mohanson** [comment]
> We don't need "v2": just replace origina one.

OK

**Review by XuJiandong — CHANGES_REQUESTED** [review]

File: contracts/ckb-combine-lock-v2/src/entry.rs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    } else {
+        Ok(ckb_std::env::argv()[0].to_bytes().into())
+    }
+}
+
+fn parse_execution_witness_args() -> Result<WitnessArgs, Error> {
+    if ckb_std::env::argv().len() == 0 {
+        Ok(load_witness_args(0, Source::GroupInput)?)
+    } else {
+        let data = ckb_std::env::argv()[1].to_bytes();
```
Comment (XuJiandong) [inline contracts/ckb-combine-lock-v2/src/entry.rs:28]: len() should be 2

File: contracts/ckb-combine-lock-v2/src/entry.rs [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        match WitnessArgsReader::verify(&data, false) {
+            Ok(()) => Ok(WitnessArgs::new_unchecked(data.into())),
```
Comment (XuJiandong) [inline contracts/ckb-combine-lock-v2/src/entry.rs:30]: This argument should be a lock field of witness, not WitnessArgs. Document updated:
> Please note, every element in inner_wintess, actually is a piece of data from lock field of WitnessArgs.

File: contracts/ckb-combine-lock-v2/src/entry.rs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                &[
+                    CString::new(child_script_args.as_str()).unwrap().as_c_str(),
+                    CString::new(child_script_inner_witness.as_str()).unwrap().as_c_str(),
+                ],
+                8,
+                &mut Vec::new(),
+            )?;
+            info!("spawn exit={}", spawn_ret);
+            if spawn_ret != 0 {
+                return Err(Error::UnlockFailed);
```
Comment (XuJiandong) [inline contracts/ckb-combine-lock-v2/src/entry.rs:90]: It's better to add `warn!` here.

**Review by XuJiandong — CHANGES_REQUESTED** [review]

**Review by XuJiandong — CHANGES_REQUESTED** [review]

File: contracts/ckb-combine-lock-v2/src/entry.rs [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+use core::result::Result;
+use log::info;
+
+fn parse_execution_args() -> Result<Bytes, Error> {
+    if ckb_std::env::argv().len() == 0 {
+        let script = load_script()?;
+        return Ok(script.args().unpack());
+    }
+    if ckb_std::env::argv().len() == 2 {
+        return Ok(ckb_std::env::argv()[0].to_bytes().into());
```
[the remark this answers is not in the document]
Comment (mohanson) [reply contracts/ckb-combine-lock-v2/src/entry.rs:21]: Right

**Review by XuJiandong — CHANGES_REQUESTED** [review]

**Review by XuJiandong — APPROVED** [review]

**Pull request merged**